homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -236,66 +272,64 @@ def _setup_from_config(self, config):
                 CONF_ERROR_TEMPLATE,
                 CONF_FAN_SPEED_TEMPLATE,
             )
+            if key in config
         }
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
-        for tpl in self._templates.values():
-            if tpl is not None:
-                tpl = MqttValueTemplate(tpl, entity=self)
 
         @callback
         @log_messages(self.hass, self.entity_id)
-        def message_received(msg):
+        def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT message."""
             if (
                 msg.topic == self._state_topics[CONF_BATTERY_LEVEL_TOPIC]
-                and self._templates[CONF_BATTERY_LEVEL_TEMPLATE]
+                and CONF_BATTERY_LEVEL_TEMPLATE in self._config
             ):
-                battery_level = self._templates[
+                battery_level: ReceivePayloadType = self._templates[
```

Comment:
Are the `ReceivePayloadType` annotations in this function necessary? After all `self._templates` has been annotated to be a dict of functions that always return them?

Reply:
I think this still applies